integrations/examples/host-functions.md

```diff
@@ -0,0 +1,152 @@
+---
+description: >-
+  A WASM module can import entities, like functions, memories, globals and
+  tables. This example illustrates how to expose functions from the host.
+---
+
+# Exposing host functions
+
+{% hint style="warning" %}
+TODO: Write this section
+{% endhint %}
+
+{% hint style="success" %}
+**Note**: The final code for this example can be found on [GitHub](https://github.com/wasmerio/wasmer/blob/master/examples/imports_function.rs).
+{% endhint %}
+
+Up until now, our WebAssembly program has only been able to do pure computation, that is, take arguments and return values. Most interesting use cases require more than just computation though. In this section we'll go over how to give the WASM modules we run extra abilities in the form of host functions.
+
+In this example, we'll create a system for getting and adjusting a counter value. However, host functions are not limited to storing data outside of WASM, they're normal host functions and can do anything that the host can do.
+
+1. There will be a `get_counter` function that will return an `i32` of
+
+   the current global counter.
+
+2. There will be an `add_to_counter` function will add the passed
+
+   `i32` value to the counter, and return an `i32` of the current
+
+   global counter.
+
+Let's generate a new project, and update our `src/main.rs` to look something like this:
+
+```rust
+use std::{cell::RefCell, sync::Arc};
+
+// Import the wasmer runtime so we can use it
+use wasmer::{imports, Function, Instance, Module, NativeFunc, Store};
+use wasmer_compiler_cranelift::Cranelift;
+use wasmer_engine_jit::JIT;
```

Comment:
In this examples we should use the default store, unless we need to use not the default one (llvm + native, for example)